src/components/restore/RestoreIcloudStep.js

```diff
@@ -0,0 +1,299 @@
+  blurOnSubmit: false,
+  letterSpacing: 0.2,
+  secureTextEntry: true,
+  size: 'large',
+  weight: 'normal',
+})`
+  padding-left: 19;
+  padding-right: 40;
+  padding-top: 15;
+  padding-bottom: 15;
+`;
+
+const IconWrapper = styled(View)`
+  margin-bottom: 12;
+  height: 22;
+  position: absolute;
+  right: 12;
+  top: 12;
+  width: 22;
+`;
+
+const Title = styled(Text).attrs({
+  size: 'big',
+  weight: 'bold',
+})`
+  margin-bottom: 12;
+`;
+
+const DescriptionText = styled(Text).attrs({
+  align: 'center',
+  color: colors.alpha(colors.blueGreyDark, 0.5),
+  lineHeight: 'looser',
+  size: 'large',
+})`
+  padding-bottom: 30;
+  padding-left: 50;
+  padding-right: 50;
+`;
+
+const WarningIcon = () => (
+  <IconWrapper>
+    <Icon color={colors.yellowOrange} name="warningCircled" size={22} />
+  </IconWrapper>
+);
+
+const TopIcon = () => (
+  <GradientText
+    align="center"
+    angle={false}
+    colors={['#FFB114', '#FF54BB', '#00F0FF']}
+    end={{ x: 0, y: 0 }}
+    letterSpacing="roundedTight"
+    size={52}
+    start={{ x: 1, y: 1 }}
+    steps={[0, 0.5, 1]}
+    weight="bold"
+  >
+    <Text size={52}>􀙶</Text>
+  </GradientText>
+);
+
+const RestoreIcloudStep = ({ userData }) => {
+  const { goBack, replace } = useNavigation();
+  const dispatch = useDispatch();
+  const { isWalletLoading } = useWallets();
+  const initializeWallet = useInitializeWallet();
+  const { accountAddress } = useAccountSettings();
+  const [validPassword, setValidPassword] = useState(false);
+  const [incorrectPassword, setIncorrectPassword] = useState(false);
+  const [passwordFocused, setPasswordFocused] = useState(true);
+  const [password, setPassword] = useState('');
+  const [label, setLabel] = useState('􀎽 Confirm Backup');
+  const passwordRef = useRef();
+  const { setComponent, hide } = usePortal();
+
+  useEffect(() => {
+    if (isWalletLoading) {
+      setComponent(
+        <LoadingOverlay
+          paddingTop={sheetVerticalOffset}
+          title={isWalletLoading}
+        />,
+        false
+      );
+    }
+    return hide;
+  }, [hide, isWalletLoading, setComponent]);
+
+  useEffect(() => {
+    const fetchPasswordIfPossible = async () => {
+      const pwd = await fetchBackupPassword();
+      if (pwd) {
+        setPassword(pwd);
+      }
+    };
+    fetchPasswordIfPossible();
+  }, []);
+
+  const onPasswordFocus = useCallback(() => {
+    setPasswordFocused(true);
+  }, []);
+
+  const onPasswordBlur = useCallback(() => {
+    setPasswordFocused(false);
+  }, []);
+
+  useEffect(() => {
+    let newLabel = '';
+    let passwordIsValid = false;
+
+    if (incorrectPassword) {
+      newLabel = 'Incorrect Password';
+    } else {
+      if (isCloudBackupPasswordValid(password)) {
+        passwordIsValid = true;
+      }
+
+      newLabel = `􀑙 Restore from iCloud`;
+    }
+
+    setValidPassword(passwordIsValid);
+    setLabel(newLabel);
+  }, [incorrectPassword, password, passwordFocused]);
```

Comment:
Do you care about passwordFocused in this dependency list? (not used in the useEffect, I think it should be covered by changes to password)